### ISSUE TITLE
Type inference failure in `sorted`

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#313]
**Describe the bug**

```python
from typing import List

a: List[str] = ["a", "bc"]
b: List[str] = sorted(a, key=len)
```

Causes an error: 

```
4:16 - error: Expression of type 'List[Union[str, Sized]]' cannot be assigned to declared type 'List[str]'
  Type 'Sized' cannot be assigned to type 'str'
    'Sized' is incompatible with 'str'
```

**version**

1.0.81

**Comment by erictraut** [comment]
Thanks for the bug report. I'm able to repro the problem, and I'll look into.

**Comment by erictraut** [comment]
The fix will be in the next published version of pyright.

**Comment by erictraut** [comment]
This is fixed in version 1.0.83, which I just published.
